net/udprelay/server.go

```diff
@@ -450,6 +452,25 @@ func (c *singlePacketConn) WriteBatchTo(buffs [][]byte, addr netip.AddrPort, gen
 	return nil
 }
 
+// UDP socket read/write buffer size (7MB). At the time of writing this value
+// was heavily influenced by magicsock, with similar motivations for its
+// increase relative to typical defaults, e.g. long fat networks and reducing
+// packet loss around crypto/syscall-induced delay.
+const socketBufferSize = 7 << 20
+
+func trySetSocketBuffer(pconn nettype.PacketConn, logf logger.Logf) {
+	directions := []sockopts.BufferDirection{sockopts.ReadDirection, sockopts.WriteDirection}
+	for _, direction := range directions {
+		errForce, errPortable := sockopts.SetBufferSize(pconn, direction, socketBufferSize)
+		if errForce != nil {
+			logf("[warning] failed to force-set UDP %v buffer size to %d: %v; using kernel default values (impacts throughput only)", direction, socketBufferSize, errForce)
+		}
+		if errPortable != nil {
+			logf("failed to set UDP %v buffer size to %d: %v", direction, socketBufferSize, errPortable)
```

Comment:
Can both of these fail, and if so does it make sense to log them both separately? (Not a huge deal but we already log SO MUCH)